agency/server-results-processing.go

```diff
@@ -22,6 +23,18 @@ func (agentState *GeneralAgentInfo) ioRequestsProcessing() {
 				defer func() {
 					<-maxIoRequestsChan
 				}()
+
+				if message.Role == engines.ChatRoleSystem {
+					systemMessage := &engines.Message{
+						ID: message.ID,
+						// ReplyTo: map[string]struct{}{correlationId: {}},
+						ReplyTo: map[string]struct{}{}, // Set to an empty map
+						Role:    engines.ChatRoleSystem,
+						Content: message.Content,
+					}
+					agentState.systemWriterChannel <- systemMessage
+					return
+				}
```

Comment:
What is the purpose of creating a new message here? Why can't we use original `message` variable?

Reply:
all really used systemMessage eventually make their way to `ioRequestsProcessing()`, so maybe it isn't necessary to send the message at agency/general-agent.go:147?

Reply:
and answering my previous comment, chances just doing `message.ReplyTo = map[string]struct{}{}` will do the job?